### PR TITLE
Fixes #266 - problem with reactimate, adds dswitch to dunai

### LINKED ISSUE #266: Bearriver: reactimate thows undefined exception on correct input
The code introduced at commit 2f21bff05a323a1f1f33558871e6ffcc73383220 doesn't seem correct.

If the first output of senseRest is a Nothing then there is an error:
```Haskell
import Data.IORef
import FRP.Yampa

main = do
  ref <- newIORef (0.0::Double)
  reactimate senseI (senseRest ref) (actuate ref) sf

senseI = do
  putStrLn "SenseI"
  return 45

senseRest ref _ = do
  v <- readIORef ref
  putStrLn "SenseRest"
  return (0.1, Nothing)

actuate ref _ x = do
  v <- readIORef ref
  putStrLn $ "actuate " ++ show x
  writeIORef ref (v+1)
  return (v >= 1)

sf = identity
```

```
SenseI
actuate 45
SenseRest
dunai-examples-bearriver-bouncingball: Prelude.undefined
CallStack (from HasCallStack):
  error, called at libraries/base/GHC/Err.hs:79:14 in base:GHC.Err
  undefined, called at src/FRP/BearRiver.hs:573:55 in bearriver-0.13.3-5dJQzIdyKtuHMkt9J9QooS:FRP.BearRiver
```

### PR DESCRIPTION
**Pull request by miguel-negrao**
Example code that caused error before this commit:

```Haskell
import Data.IORef
import FRP.Yampa

main = do
  ref <- newIORef (0.0::Double)
  reactimate senseI (senseRest ref) (actuate ref) sf

senseI = do
  putStrLn "SenseI"
  return 45

senseRest ref _ = do
  v <- readIORef ref
  putStrLn "SenseRest"
  return (0.1, Nothing)

actuate ref _ x = do
  v <- readIORef ref
  putStrLn $ "actuate " ++ show x
  writeIORef ref (v+1)
  return (v >= 1)

sf = identity
```

### CODE REVIEW

File: dunai/src/Control/Monad/Trans/MSF/Except.hs [hunk 1]
```diff
@@ -273,6 +273,14 @@ switch sf f = catchS ef  f
            (b,me)  <- liftTransS sf -< a
            inExceptT -< ExceptT $ return $ maybe (Right b) Left me
 
+-- * Analog to Yampa's dswitch, with Maybe instead of Event
```
[the remark this answers is not in the document]
Comment (miguel-negrao) [reply dunai/src/Control/Monad/Trans/MSF/Except.hs:276]: I believe this is just comment, not haddock, right ? I copy-pasted from switch.

[the remark this answers is not in the document]
Comment (miguel-negrao) [reply dunai/src/Control/Monad/Trans/MSF/Except.hs]: Changed to haddock.

**Comment by ivanperez-keera** [comment]
An issue has just been closed to address the fact that the haddock syntax in `switch` was the section syntax, not the function documentation syntax.

That's independent from this issues.

@miguel-negrao Could you please rebase this on top of the current `develop`? It will give you a conflict, but it will also reduce the size of the commit (by 1-2 lines).

Also, please see my questions regarding the syntax (two are requests for types, the other is a syntax question).

**Comment by miguel-negrao** [comment]
@ivanperez-keera I can't seem to find your questions regarding the syntax. Are they comments in this PR ?

I've rebased on developed. I've also simplified switch's implementation using `throwMaybe`. I believe the code becomes clearer with this change, but check if you feel the same.

**Comment by ivanperez-keera** [comment]
Thanks!

I started it as a review of your code. The questions were attached to the implementation:

Screenshot from 2022-03-23 11-10-07

**Comment by miguel-negrao** [comment]
I think this happened to me once. I think you have to hit something like "submit review" for the review to go live and become available.

**Comment by ivanperez-keera** [comment]
It's ok. You've answered the question, so that's all I needed. Thanks!

**Comment by ivanperez-keera** [comment]
I cannot replicate the issue. I've done some work on this issue and left a comment here: https://github.com/ivanperez-keera/dunai/issues/266#issuecomment-1086673235

**Comment by ivanperez-keera** [comment]
This PR will be closed but some changes have been merged and @miguel-negrao credited for them.

See https://github.com/ivanperez-keera/dunai/issues/266#issuecomment-1086705215 for comments regrading the regression test and #276 for a follow-up issue on the simplification of switch.

Thanks!

**Pull request closed without merging**